cpp/core/operators/shuffle/splitter.h

```diff
@@ -56,7 +57,7 @@ class Splitter {
    * partition id. The largest partition buffer will be spilled if memory
    * allocation failure occurs.
    */
-  virtual arrow::Status Split(const arrow::RecordBatch&);
+  virtual arrow::Status Split(std::shared_ptr<ColumnarBatch> cb);
```

Comment:
use `ColumnarBatch&` or `ColumnarBatch*` instead.
If no modification is made to `cb` in the function, you can add `const` qualifier.

`std::shared_ptr` means shared semantics, refer to [R.30](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rr-smartptrparam)